libs/designsystem/src/lib/components/modal/services/modal.helper.spec.ts

```diff
@@ -114,6 +147,133 @@ describe('ModalHelper', () => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            flavor: 'drawer',
+            component: InputEmbeddedComponent,
+          });
+          ionModal = await ionModalController.getTop();
+          expect(ionModal).toBeTruthy();
+          expect(ionModal.classList.contains('small')).toBeFalse();
+          expect(ionModal.classList.contains('medium')).toBeFalse();
+          expect(ionModal.classList.contains('large')).toBeFalse();
+          await overlay.dismiss();
+        });
+
+        it("should add class `content-overflows`, if content can't fit in viewport", async () => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            component: ContentOverflowsWithFooterEmbeddedComponent,
+          });
+          ionModal = await ionModalController.getTop();
+          const ionModalWrapper = ionModal.querySelector('.modal-wrapper');
+          expect(ionModal).toBeTruthy();
+          expect(ionModalWrapper).toBeTruthy();
+
+          await TestHelper.whenTrue(() => {
+            return ionModalWrapper.classList.contains('content-overflows');
+          });
+
+          expect(ionModalWrapper.classList.contains('content-overflows')).toBeTrue();
+          await overlay.dismiss();
+        });
+
+        it('should NOT add class `content-overflows`, if content can fit in viewport', async (done) => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            component: ContentWithNoOverflowEmbeddedComponent,
+          });
+          ionModal = await ionModalController.getTop();
+          const ionModalWrapper = ionModal.querySelector('.modal-wrapper');
+          expect(ionModal).toBeTruthy();
+          expect(ionModalWrapper).toBeTruthy();
+
+          setTimeout(async () => {
+            expect(ionModalWrapper.classList.contains('content-overflows')).toBeFalse();
+            await overlay.dismiss();
+            done();
+          });
+        });
+
+        it('should have footer visible at the bottom of viewport, when content-overflows', async () => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            component: ContentOverflowsWithFooterEmbeddedComponent,
+          });
+          ionModal = await ionModalController.getTop();
+          const ionModalWrapper = ionModal.querySelector('.modal-wrapper');
+          const footer = ionModal.querySelector('kirby-modal-footer');
+          expect(ionModal).toBeTruthy();
+          expect(ionModalWrapper).toBeTruthy();
+          expect(footer).toBeTruthy();
+
+          await TestHelper.whenTrue(() => {
+            return ionModalWrapper.classList.contains('content-overflows');
+          });
+
+          expect(ionModalWrapper.classList.contains('content-overflows')).toBeTrue();
+          expect(footer.getBoundingClientRect().bottom).toEqual(window.innerHeight);
+          await overlay.dismiss();
```

Comment:
Consider moving this to `afterEach` to ensure modal is dismissed if `expect` throws an error